codes/codes.go

```diff
@@ -0,0 +1,172 @@
+package codes
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type Code uint32
+
+const (
+	// OK is returned on success.
+	OK Code = 0
+
+	// Canceled indicates the operation was canceled (typically by the caller).
+	Canceled Code = 1
+
+	// Unknown error. An example of where this error may be returned is
+	// if a Status value received from another address space belongs to
+	// an error-space that is not known in this address space. Also
+	// errors raised by APIs that do not return enough error information
+	// may be converted to this error.
+	Unknown Code = 2
+
+	// InvalidArgument indicates client specified an invalid argument.
+	// Note that this differs from FailedPrecondition. It indicates arguments
```

Comment:
Several of these comments refer to `FailedPrecondition` but that is not one of the errors brought over. Can you update the comments or add `FailedPrecondition` back in?

Reply:
I'm going to look at `FailedPrecondition` again. It seems to be mentioned a lot and I didn't understand it at first. I'll update these comments regardless, but I want a better reason why I think `FailedPrecondition` shouldn't be there other than, "I didn't understand what it meant and it didn't seem to apply to any situations I could think of."